sw/device/silicon_creator/lib/bootstrap.h

```diff
@@ -4,12 +4,30 @@
 #ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_BOOTSTRAP_H_
 #define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_BOOTSTRAP_H_
 
+#include <stdint.h>
+
+#include "sw/device/lib/base/hardened.h"
+#include "sw/device/silicon_creator/lib/drivers/spi_device.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#include "sw/device/lib/base/hardened.h"
-#include "sw/device/silicon_creator/lib/error.h"
+/**
+ * Determines whether the given SPI opcode is allowed to operate on the given
+ * page.
+ *
+ * NOTE: Code that depends on this library must provide an implementation of
+ * this function.
+ *
```

Comment:
@alphan Do you have thoughts on this style of dependency injection? My motivation was to remove some ROM_EXT-specific logic from the common library, but I'm not sure whether it's worth the additional complexity.

Reply:
Per our chat offline, it sounds like this style is fine, in fact you'd like to see even more if it! I'll work on refactoring this PR a bit to eliminate the `bounds_checks_on` parameter (and the dead code that results from it).